File: 523_Continuous_Subarray_Sum.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        visited = {0: -1}
        subsum = 0
        for i in range(len(nums)):
            subsum = (subsum + nums[i]) % k
            if subsum in visited:
                if i - visited[subsum] > 1:
                    return True
            else:
                visited[subsum] = i
        return False

File: 523_Continuous_Subarray_Sum.go
package main

func checkSubarraySum(nums []int, k int) bool {
	visited := map[int]int{0: -1}
	subSum := 0
	for i, num := range nums {
		subSum = (subSum + num) % k
		if v, ok := visited[subSum]; ok {
			if i-v > 1 {
				return true
			}
		} else {
			visited[subSum] = i
		}
	}
	return false
}
